statistics/histogram.go

```diff
@@ -1186,14 +1186,70 @@ func (idx *Index) GetRowCount(sc *stmtctx.StatementContext, indexRanges []*range
 		}
 		l := types.NewBytesDatum(lb)
 		r := types.NewBytesDatum(rb)
-		totalCount += idx.BetweenRowCount(l, r)
 		lowIsNull := bytes.Equal(lb, nullKeyBytes)
 		if (idx.outOfRange(l) && !(isSingleCol && lowIsNull)) || idx.outOfRange(r) {
 			totalCount += outOfRangeEQSelectivity(outOfRangeBetweenRate, modifyCount, int64(idx.TotalRowCount())) * idx.TotalRowCount()
 		}
 		if isSingleCol && lowIsNull {
 			totalCount += float64(idx.NullCount)
 		}
+		// Due to the limitation of calcFraction and convertDatumToScalar, the histogram actually won't estimate anything.
+		// If the first column's range is point.
+		if rangePosition := GetOrdinalOfRangeCond(sc, indexRange); rangePosition > 0 && idx.StatsVer == Version2 && coll != nil {
+			tmpRan := []*ranger.Range{
+				{
+					LowVal:  make([]types.Datum, 1),
+					HighVal: make([]types.Datum, 1),
+				},
+			}
+			colsIDs := coll.Idx2ColumnIDs[idx.ID]
+			singleColumnEstResults := make([]float64, 0, len(indexRange.LowVal))
+			// The following codes are the same with SQLServer's CSelCalcCombineFilters_ExponentialBackoff Calculator. It works like:
```

Comment:
How about removing keywords `SQLServer` and `CSelCalcCombineFilters_ExponentialBackoff`? 
I don't know if there is any legal risk since `SQLServer` is not open-sourced. But for safety, I think it's better to remove them.

Reply:
But how we reference the original source?

Reply:
@zz-jason Any suggestion?